### ISSUE TITLE
Using a plugin dependency only to compile test groovy code

### ISSUE THREAD

**Issue by khmarbaise** [groovy/GMavenPlus#86]
I have some groovy code located in `src/test/groovy/...` and configured `gmavenplus-maven-plugin` using the goal `compileTests` to compile this code only...so the code should be compiled.

So I have to define the dependency globally like this:

```xml
<dependency>
  <groupId>org.codehaus.groovy</groupId>
  <artifactId>groovy-all</artifactId>
  <version>2.4.12</version>
  <scope>test</scope>
</dependency>
```

It does not work to define the dependency as a dependency of the plugin itself like this:
```xml
<plugins>
  <plugin>
    <groupId>org.codehaus.gmavenplus</groupId>
    <artifactId>gmavenplus-plugin</artifactId>
    <version>1.6</version>
    <dependencies>
      <dependency>
        <groupId>org.codehaus.groovy</groupId>
        <artifactId>groovy-all</artifactId>
        <version>2.4.12</version>
      </dependency>
    </dependencies>
  </plugin>
</plugins>
```
Maybe I oversight an option to define the dependency only on the plugin level instead of the global level ?

**Comment by keeganwitt** [comment]
You're not missing anything, that indeed doesn't work.  I intentionally didn't include the plugin's classpath on the project's compilation classpath so that the plugin's dependencies don't override the versions your project declares.

GMavenPlus doesn't have a ton of dependencies though, the other day I was thinking about adding an option to use the plugin's classloader (which will include the plugin's dependencies, so you can do as you are asking), then you could override any plugin dependencies that conflict between the plugin and the project to ensure you compile against the version you want.  This option would have to used carefully.

Out of curiosity, what is your use case?  Your project will need Groovy at test runtime anyway, right?

**Comment by khmarbaise** [comment]
I'm using the [groovy part for integration tests via maven-invoker-plugin only](https://github.com/khmarbaise/multienv-maven-plugin/blob/gmavenplus-maven-plugin/pom.xml)...so it would be convenient to use that only as a plugin dependency...cause this would give the option to only make a configuration in a parent and just add a line for execution in the targeting plugin (for example in multienv-maven-plugin in my case)...and the dependency which is really needed is already configured with the pluginManagement...otherwise I have to add an explicit dependency in my targeting project or add a dependency for all inherited projects...

**Comment by keeganwitt** [comment]
So for your use case, you never invoke those tests through your IDE?  Because that would require Groovy to be on the classpath with your integration tests to execute, right?

**Comment by khmarbaise** [comment]
Yes correctly. Only writing them via IDE ...can't execute them via IDE cause they need Maven has ran...

**Comment by keeganwitt** [comment]
One thing I was noticing while I was playing around with this  idea -- Having Groovy as a plugin dependency won't help Surefire execute your tests -- Groovy would still need to be a project dependency for that to work.  Given that's the case, I wonder if adding this feature would do any good.

**Comment by keeganwitt** [comment]
I'm still not sure this is a good idea, but I merged a change for #64 that was asking to do this for GroovyDoc, and as part of that I added the ability to use the same classloader to load both the project dependencies and plugin dependencies for any goal you choose (including compilation).
